src/Compilers/CSharp/Portable/Symbols/MemberSignatureComparer.cs

```diff
@@ -380,38 +501,52 @@
+        // Note: the order of named arguments is ignored
+        private readonly bool _considerAttributes;
+
         private MemberSignatureComparer(
             bool considerName,
             bool considerExplicitlyImplementedInterfaces,
             bool considerReturnType,
             bool considerTypeConstraints,
             bool considerCallingConvention,
             RefKindCompareMode refKindCompareMode,
-            bool considerArity = true,
-            bool considerDefaultValues = false,
-            TypeCompareKind typeComparison = TypeCompareKind.IgnoreDynamic | TypeCompareKind.IgnoreNativeIntegers)
+            bool considerTypeParameterAndParameterNames,
+            bool considerAttributes,
+            bool considerTypeParameters,
+            bool considerDefaultValues,
+            TypeCompareKind typeComparison)
         {
             Debug.Assert(!considerExplicitlyImplementedInterfaces || considerName, "Doesn't make sense to consider interfaces separately from name.");
-            Debug.Assert(!considerTypeConstraints || considerArity, "If you consider type constraints, you must also consider arity");
+            Debug.Assert(!considerTypeConstraints || considerTypeParameters, "If you consider type constraints, you must also consider type parameters");
 
             _considerName = considerName;
             _considerExplicitlyImplementedInterfaces = considerExplicitlyImplementedInterfaces;
             _considerReturnType = considerReturnType;
             _considerTypeConstraints = considerTypeConstraints;
             _considerCallingConvention = considerCallingConvention;
             _refKindCompareMode = refKindCompareMode;
-            _considerArity = considerArity;
+            _considerTypeParameters = considerTypeParameters;
             _considerDefaultValues = considerDefaultValues;
             _typeComparison = typeComparison;
             Debug.Assert((_typeComparison & TypeCompareKind.FunctionPointerRefMatchesOutInRefReadonly) == 0,
                          $"Rely on the {nameof(refKindCompareMode)} flag to set this to ensure all cases are handled.");
-            Debug.Assert(_refKindCompareMode == RefKindCompareMode.DoNotConsiderDifferences ||
+            Debug.Assert(_refKindCompareMode == RefKindCompareMode.RefMatchesOutInRefReadonly ||
+                _refKindCompareMode == RefKindCompareMode.IgnoreRefKind ||
                 (_refKindCompareMode & RefKindCompareMode.ConsiderDifferences) != 0,
                 $"Cannot set {nameof(RefKindCompareMode)} flags without {nameof(RefKindCompareMode.ConsiderDifferences)}.");
-            if ((refKindCompareMode & RefKindCompareMode.ConsiderDifferences) == 0)
+
+            if (refKindCompareMode == RefKindCompareMode.RefMatchesOutInRefReadonly)
             {
                 _typeComparison |= TypeCompareKind.FunctionPointerRefMatchesOutInRefReadonly;
             }
+
+            _considerParameterNames = considerTypeParameterAndParameterNames;
```

Comment:
>_considerParameterNames [](http://example.com/codeflow?start=12&length=23)

Consider renaming this field as well.

Reply:
You may be looking at an older commit. I updated the field later on